packages/components/src/slot-fill/test/observable-map.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { render, screen, act } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import {
+	observableMap,
+	useObservableValue,
+} from '../bubbles-virtually/observable-map';
+
+describe( 'ObservableMap', () => {
+	test( 'should observe individual values', () => {
+		const map = observableMap();
+		let notifsFromA = 0;
+		let notifsFromB = 0;
+
+		const unsubA = map.subscribe( 'a', () => {
+			notifsFromA++;
+		} );
+		const unsubB = map.subscribe( 'b', () => {
+			notifsFromB++;
+		} );
+
+		// check that setting `a` doesn't notify the `b` listener
+		map.set( 'a', 1 );
+		expect( notifsFromA ).toBe( 1 );
+		expect( notifsFromB ).toBe( 0 );
+
+		// check that setting `b` doesn't notify the `a` listener
+		map.set( 'b', 2 );
+		expect( notifsFromA ).toBe( 1 );
+		expect( notifsFromB ).toBe( 1 );
+
+		// check that `delete` triggers notifications, too
+		map.delete( 'a' );
+		expect( notifsFromA ).toBe( 2 );
+		expect( notifsFromB ).toBe( 1 );
+
+		// check that the subscription survived the `delete`
+		map.set( 'a', 2 );
+		expect( notifsFromA ).toBe( 3 );
+		expect( notifsFromB ).toBe( 1 );
+
+		// check that unsubscription really works.
+		unsubA();
+		unsubB();
+		map.set( 'a', 3 );
+		expect( notifsFromA ).toBe( 3 );
+		expect( notifsFromB ).toBe( 1 );
+	} );
+} );
+
+describe( 'useObservableValue', () => {
+	test( 'reacts only to the specified key', () => {
+		const map = observableMap();
+		map.set( 'a', 1 );
+
+		let renders = 0;
```

Comment:
Perhaps a cleaner way to store the count inside a ref in the component?

We could then render that value and assert it the same way we assert "value is N"

Reply:
Yes, but this works equally well, it's not complicated or confusing in anyway, and we're already doing this in other tests, like for `useSelect`.

Reply:
It kind of feels off to me to pass and modify an external variable in a component, but it's not critical in a test. Leaving it to your judgment.

Reply:
I wrapped the functions where we count how many times they are called in a `jest.fn`, and started checking the call counts with `.toHaveBeenCalledTimes`. That should be a more professional way to do it 🙂